src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -756,54 +796,47 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
     // - a http_request for the client to send
     // Return value:
     // - the response from the server as a json value
-    json::value AzureConnection::_SendRequestReturningJson(http_client& theClient, http_request theRequest)
+    WDJ::JsonObject AzureConnection::_SendRequestReturningJson(std::wstring_view uri, const WWH::IHttpContent& content, WWH::HttpMethod method)
     {
-        auto& headers{ theRequest.headers() };
-        headers.add(L"User-Agent", HttpUserAgent);
-        headers.add(L"Accept", L"application/json");
-
-        json::value jsonResult;
-        const auto responseTask = theClient.request(theRequest);
-        responseTask.wait();
-        const auto response = responseTask.get();
-        const auto responseJsonTask = response.extract_json();
-        responseJsonTask.wait();
-        jsonResult = responseJsonTask.get();
+        if (!method)
+        {
+            method = content == nullptr ? WWH::HttpMethod::Get() : WWH::HttpMethod::Post();
+        }
+
+        WWH::HttpRequestMessage request{ method, Uri{ uri } };
+        request.Content(content);
+
+        auto headers{ request.Headers() };
+        headers.Accept().TryParseAdd(L"application/json");
+
+        auto response{ _httpClient.SendRequestAsync(request).get() };
+        auto string{ response.Content().ReadAsStringAsync().get() };
+        auto jsonResult{ WDJ::JsonObject::Parse(string) };

```

Comment:
const

probably on some of the other ones too

Reply:
const doesn't mean anything to winrt classes, fwiw, but you're right.